### ISSUE TITLE
Highlighting of keys in dictionary literals when using the official Python syntax definition

### ISSUE THREAD

**Issue by vaultah** [MattDMo/Neon-color-scheme#49]
If I use the Neon color scheme (version 3.2.2) and Sublime Text's official Python syntax definition (build 4126), I get dictionary literals highlighted like this:

![Screenshot from 2022-03-23 22-52-24](https://user-images.githubusercontent.com/4944562/159809298-65ddf690-7612-417d-b834-6f963bc5322f.png)

This looks somewhat confusing, because the only distinction between constant and non-constant parts of expressions used as keys are quotes, which are tiny and too easy to miss.

If I use Sublime Text's built-in color schemes (like Monokai), dictionary keys are highlighted as expected:

![Screenshot from 2022-03-23 22-58-39](https://user-images.githubusercontent.com/4944562/159810186-ea19bd40-3b33-4767-a4d2-399d4854bcaf.png)

I'm not exactly sure where the problem is. FWIW I never noticed this problem while using the MagicPython syntax definition, but it's not listed as supported by Neon color scheme.

**Comment by MattDMo** [comment]
The way it works now is that a dict key, regardless of whether it's a variable or string literal, is colored light green. A value is actually supposed to be light blue as well, for variables and string literals, but it looks like I missed defining a color for variables. I wanted to clearly differentiate between keys and values, regardless of what type of entity they are. From what you're saying, you'd like to have strings differentiated from variables, right?

Let me see if I can come up with a way to keep a visually similar, but not identical, element to all keys and all values, yet still have a difference between string literals and variables.

**Comment by vaultah** [comment]
> From what you're saying, you'd like to have strings differentiated from variables, right?

Yeah, I would just want string literals and a variables in the key portion to be more easily distinguishable. I often prefer single quotes, which are even tinier than the double quotes in the example above, so variable and constant parts of dictionary keys visually blend together. Here's a real world example:

![Screenshot from 2022-03-25 15-08-33](https://user-images.githubusercontent.com/4944562/160147579-432a2020-4c61-41f9-ae98-895f5540b6e8.png)

They don't have to be highlighted exactly like with other color schemes. I actually really like this color and the way it looks with bold values :)

**Comment by vaultah** [comment]
FWIW I want to switch from MagicPython syntax definition to the official one, but I guess MagicPython does something different in terms of scopes, causing this color scheme to highlight keys more in line with my expectations:

![Screenshot from 2022-03-25 15-20-41](https://user-images.githubusercontent.com/4944562/160149738-8efaeed2-91e2-459e-8370-8e7cc6820e37.png)

**Comment by MattDMo** [comment]
OK, I see what you mean, thanks!

I've never really used MagicPython. A number of years ago the regular Python syntax was pretty bad, so in response I wrote [`Python Improved`](https://packagecontrol.io/packages/Python%20Improved). Unfortunately, development on that has lagged for a couple of reasons - declining free time on my part, and fantastic improvements to the stock Python syntax after the Packages were open-sourced. It now is only missing one feature that I want - highlighting of `BUG`/`TODO`/`XXX` in comments. I'll probably put in a PR at some point.

**Comment by MattDMo** [comment]
How does this look?

![image](https://user-images.githubusercontent.com/1776358/160161542-2953757b-7bee-4f33-8939-bb1e182d94dc.png)

The key coloring is still there, but there's a muted background for string literals. 

Or there's this:

![image](https://user-images.githubusercontent.com/1776358/160161877-e9757cbb-fb17-4ccc-a62a-0cf502c20674.png)

with a more muted background.

**Comment by vaultah** [comment]
Both options are already perfectly acceptable, though I personally prefer the second (more muted) version.

Just a thought, would it look slightly better if this background was expanded to the surrounding quotes as well? Similar to the way curly braces in `{placeholders}` in f-strings have the same background as the enclosed part. Also, I'd say the same kind of background (but blue-ish) could be added to the string literals in the _value_ portion as well (for symmetry).

But I can definitely live with the proposed solution, so the rest is completely up to you :P

**Comment by vaultah** [comment]
Or, rather, it might be better to add this background to _variables_ and leave string literals as is -- now that I think about it, the absolute majority of dictionary literals that I see in the real world use _only_ string literals for keys. In other words, it'd be better to keep the style that already works well in by far the most common case and add small tweaks to improve it for less common cases
